### DISCUSSION TITLE
Problem working with multiple protocols

### DISCUSSION THREAD

**Discussion by macruz21** [synchronizing/mitm#28] [Q&A] [answered]
Hello,
First of all congratulations for your code. I found it very elegant. I am using it for several things that I have in mind, but I have found a problem in the code. When you use for to go through all the possible protocols, there should be a break when a protocol doesn't throw the "InvalidProtocol" exception, because otherwise it makes multiple connections until it reaches the last protocol that is actually executed.
In this way, when it reaches the first protocol that works, it exits the for.
Thank you very much for your code, I am enjoying a lot with it.
Greetings.

**Comment by synchronizing** [comment, marked as answer]
Created a new issue to track. Closing discussion for issue. #29
